traitlets/traitlets.py

```diff
@@ -418,6 +420,10 @@ def _dynamic_default_callable(self, obj):
         return getattr(self, 'make_dynamic_default', None)
 
     def instance_init(self, obj):
+        if not hasattr(obj, '_trait_values'):
```

Comment:
Yes, I also though of adding an instance-level method called from within `HasDescriptors.__new__` but did not want to add complexity in the `HasDescriptors` unless really required.